src/XrmContext/Interpretation/InterpretEntityMetadata.fs

```diff
@@ -74,11 +74,12 @@ let interpretAttribute deprecatedPrefix entityNames (e:EntityMetadata) (a:Attrib
       | _ -> None, false
 
     let vType = 
-      match aType, hasOptions with
-      | AttributeTypeCode.State,    true
-      | AttributeTypeCode.Status,   true
-      | AttributeTypeCode.Picklist, true -> OptionSet options.Value.displayName
-      | AttributeTypeCode.PartyList, _   -> PartyList
+      match aType, hasOptions, a with
+      | AttributeTypeCode.State,    true, _
+      | AttributeTypeCode.Status,   true, _
+      | AttributeTypeCode.Picklist, true, _ -> OptionSet options.Value.displayName
+      | AttributeTypeCode.Virtual , true, :? MultiSelectPicklistAttributeMetadata -> OptionSetCollection options.Value.displayName
```

Comment:
Adding the virtual attribute type-check to this section of the code makes it a bit too intricate, imo. As per the above comment, I would suggest handling virtual attributes separately since they are a special case.